h2o-docs/src/product/data-science/permutation-variable-importance.rst

```diff
@@ -0,0 +1,95 @@
+Permutation Variable Importance
+-------------------------------
+
+Introduction
+~~~~~~~~~~~~
+
+Permutation variable importance is obtained by measuring the distance between prediction errors before and after a feature is permuted; only one feature at a time is permuted.
+
+
+Implementation
+~~~~~~~~~~~~~~
+
+The model is scored on a dataset ``D``, this yields some metric value ``orig_metric`` for metric ``M``.
+
+Permutation variable importance of a variable ``V`` is calculated by the following process:
+
+1) Variable ``V`` is randomly shuffled using `Fisher-Yates algorithm <https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle>`__.
+2) The model is scored on the dataset ``D`` with the variable ``V`` replaced by the result from step **1.** this yields some metric value ``perm_metric`` for the same metric ``M``.
+3) Permutation variable importance of the variable ``V`` is then calculated as ``abs(perm_metric - orig_metric)``.
+
+Metric ``M`` can be set by **metric** argument. If set to ``AUTO``, ``AUC`` is used for binary classification,
+``logloss`` is used for multinomial classification, and ``RMSE`` is used for regression.
+
+Parameters
+~~~~~~~~~~
+
+- **model**: A trained model for which it will be used to score the dataset.
+- **frame**: The dataset to use, both train and test frame are can be reasonable choices but the interpretation differs (see `Should I Compute Importance on Training or Test Data? <https://christophm.github.io/interpretable-ml-book/feature-importance.html#feature-importance-data>`__ from the Interpretable Machine Learning by Christoph Molnar.).
+- **metric**: The metric to be used to calculate the error measure. One of ``AUTO``, ``AUC``, ``MAE``, ``MSE``, ``RMSE``, ``logloss``, ``mean_per_class_error``, ``PR_AUC``. Defaults to ``AUTO``.
+- **n_samples**: The number of samples to be evaluated. Use -1 to use the whole dataset. Defaults to 10 000.
+- **n_repeats**: The number of repeated evaluations. Defaults to 1.
+- **features**: The features to include in the permutation importance. Use None to include all.
```

Comment:
Should we use the term `columns` since all the rest of the column (or list of column) arguments have "column" in the name?  e.g. `fold_column`, `ignore_columns`, `gam_columns`, etc.  I don't think we have an args with "feature" in the name yet.


Looks like `n_repeats` is replicated to [match sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.inspection.permutation_importance.html), and `n_samples` is new to H2O?  I can't remember if we talked about this on Slack or not yet (were there existing arg names for something like n samples in H2O)?

Reply:
Thank you for the consistency check! 

There is a thread on api updates channel about the parameter but not about the naming. `n_samples` would be new to h2o. Now when searching I noticed we have `samples` in SHAP plot. Would it be better to use `samples` here as well?

I will change `features` to `columns` when we decide what to do with the `n_samples`.